drivers/hmis_data_quality_tool/app/controllers/hmis_data_quality_tool/warehouse_reports/reports_controller.rb

```diff
@@ -36,6 +37,27 @@ def show
       end
     end
 
+    def by_client
+      @clients = @report.clients.order(:last_name, :first_name)
+      @pagy, @clients = pagy(@clients)
+    end
+
+    private def pivot_report
+      @groups = @report.result_groups.except('Inventory')
+      @lookup = (
+        ::HmisDataQualityTool::Client.sections(@report).map { |k, v| [k, v[:title]] } +
+          ::HmisDataQualityTool::Enrollment.sections(@report).map { |k, v| [k, v[:title]] } +
+          ::HmisDataQualityTool::CurrentLivingSituation.sections(@report).map { |k, v| [k, v[:title]] }
+      ).to_h
+
+      @flags = {}
+      @groups.values.map(&:keys).flatten.each do |key|
+        question_name = "#{@lookup[key]}__invalid"
+        @flags[@lookup[key]] = @report.universe(question_name).members.pluck(:personal_id, :data_source_id)
+      end
+      @clients_with_flags = @flags.values.flatten(1).uniq
```

Comment:
This all feels like something I'd like to move into the report model, so you could call:
```
@lookup = @report.pivot_lookup
@flags = @report.pivot_flags
@clients_with_flags  @report.pivot_clients_with_flags
```

Or better:
```
@pivot_details = @report.pivot_details
```
and just have it return a hash of what you need.